numpy/core/src/common/simd/avx2/arithmetic.h

```diff
@@ -118,7 +119,16 @@
 #endif // !NPY_HAVE_FMA3
 
 // Horizontal add: Calculates the sum of all vector elements.
-NPY_FINLINE npy_uint32 npyv_sum_u32(__m256i a)
+
+NPY_FINLINE npy_uint16 npyv_sumup_u8(npyv_u8 a)
```

Comment:
Possibly worth a better name,
```suggestion
NPY_FINLINE npy_uint16 npyv_sum_u8_to_u16(npyv_u8 a)
```
but maybe this makes the templating more trouble.

Reply:
we actually follow this notation `npyv_{intrinsic}_{return type}_{parm type}` without `_to_` in other areas but this case is different because:

- sum returns a scalar rather than a vector
- we are dealing with two different behavior of summation so it's better to give it a different name. sum(reduce sum) and sumup(extend sum)
- we may need to add another intrinsics to perform saturate sum e.g. `npyv_sums_*`

EDIT: rewrote this comment due to the duplication

Reply:
you are right, `_to_` may cause an misleading return type.  we can add `npyv_sums_*` if needed.